ValveKeyValue/ValveKeyValue/ObjectCopier.cs

```diff
@@ -84,9 +84,10 @@ static KVValue ConvertObjectToValue(Type objectType, object managedObject, IObje
                 throw new KeyValueException("Serialization failed - circular object reference detected.");
             }
 
-            if (TryConvertValueTo<string>(string.Empty, managedObject, out var convertedString))
+            if (CanConvertValueTo(objectType) && managedObject is IConvertible)
             {
-                return convertedString;
+                var converter = typeof(KVValue).GetMethod("op_Implicit", new[] { objectType });
```

Comment:
@yaakov-h please don't have a heart attack here.

This *works* but is pretty insane. `Convert.ChangeType` does not want to convert to `KVValue`.

But with this change we're keeping all the types around, instead of forcing everything to become a string.

Reply:
🤨 
I'd rather do some type switching than runtime method overloading via reflection here. Perhaps a factory function on `KVValue`?

(As a curiosity only, I wonder if a `(KVValue)(dynamic)` double-cast would work here)

Reply:
Yeah type switching is an obvious way of doing it, I just didn't want to write out all the types. But in the long term I suppose that's just the better way of doing it.

I was fine-ish with reflection here because this whole object copying already uses reflection to do it's job.